src/Analyzers/CSharp/Tests/UseParameterNullChecking/UseParameterNullCheckingTests.cs

```diff
@@ -0,0 +1,1453 @@
+        public async Task TestWithAlias()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using Ex = System.ArgumentNullException;
+class C
+{
+    static void M(C c)
+    {
+        [|if ((object)c == null)
+            throw new Ex(nameof(c));|]
+    }
+}
+",
+                FixedCode = @"using Ex = System.ArgumentNullException;
+class C
+{
+    static void M(C c!!)
+    {
+    }
+}
+",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestPointer()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using System;
+class C
+{
+    static unsafe void M(int* ptr)
+    {
+        [|if (ptr == null)
+            throw new ArgumentNullException(nameof(ptr));|]
+    }
+}
+",
+                FixedCode = @"using System;
+class C
+{
+    static unsafe void M(int* ptr!!)
+    {
+    }
+}
+",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestFunctionPointer()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"using System;
+class C
+{
+    static unsafe void M(delegate*<int, void> ptr)
+    {
+        [|if (ptr == null)
+            throw new ArgumentNullException(nameof(ptr));|]
+    }
+}
+",
+                FixedCode = @"using System;
+class C
+{
+    static unsafe void M(delegate*<int, void> ptr!!)
+    {
+    }
+}
+",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestPartialMethod()
+        {
+            var partialDeclaration = @"
+partial class C
+{
+    static partial void M(C c);
+}
+";
+
+            await new VerifyCS.Test()
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        partialDeclaration,
+                        @"using System;
+partial class C
+{
+    static partial void M(C c)
+    {
+        [|if ((object)c == null)
+            throw new ArgumentNullException(nameof(c));|]
+    }
+}
+"
+                    }
+                },
+                FixedState =
+                {
+                    Sources =
+                    {
+                        partialDeclaration,
+                        @"using System;
+partial class C
+{
+    static partial void M(C c!!)
+    {
+    }
+}
+"
+                    }
+                },
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
```

Comment:
add test for:

```
public C(
      string x // comment
    , string y // comment,
    , string z // comment
    )
{
}
```

Reply:
Good test case! We actually put the `!!` on the next line here. It feels like the trailing trivia of the identifier should move over to the `!!`. Is there an existing helper that lets me "insert a token and move over another token's trivia" in this way?

Reply:
nope.  you'd have to write the logic yourself.  note: i'm also ok with you leaving that out of this PR and waiting until tehre is user feedback :)